app/controllers/locks_controller.rb

```diff
@@ -7,26 +13,39 @@ def create
       permit(:description, :stage_id, :warning).
       merge(user: current_user)
     Lock.create!(attributes)
-    redirect_to :back, notice: "Locked"
+    redirect_to :back, notice: 'Locked'
   end
 
   def destroy
     lock.try(:soft_delete)
-    redirect_to :back, notice: "Unlocked"
+    redirect_to :back, notice: 'Unlocked'
   end
 
   protected
 
   def for_global_lock?
     case action_name
-    when "create" then !params[:lock].try(:[], :stage_id)
-    when "destroy" then !lock.stage_id
+    when 'create' then
+      !params[:lock].try(:[], :stage_id) || !params[:lock].try(:[], :stage_id).presence
```

Comment:
Right. `params[:lock].try(:[], :stage_id).present?` will cover both of those cases.